File: hunt/tests/fixtures/semgrep/raw_rsa.js
const crypto = require('crypto');

function positive(key, buf) {
  const a = crypto.privateDecrypt({ key, padding: crypto.constants.RSA_NO_PADDING }, buf); // EXPECT:raw-rsa-node-no-padding-private
  const b = crypto.privateEncrypt({ key: key, padding: crypto.constants.RSA_NO_PADDING }, buf); // EXPECT:raw-rsa-node-no-padding-private
  return [a, b];
}

function negative(key, pub, buf) {
  const c = crypto.privateDecrypt({ key, padding: crypto.constants.RSA_PKCS1_OAEP_PADDING }, buf);
  const d = crypto.publicEncrypt({ key: pub, padding: crypto.constants.RSA_NO_PADDING }, buf);
  return [c, d];
}

File: hunt/tests/fixtures/semgrep/raw_rsa.c
#include <openssl/rsa.h>
#include <openssl/evp.h>
#include <pkcs11.h>

void positive(RSA *rsa, EVP_PKEY_CTX *ctx, unsigned char *in, unsigned char *out, int len) {
    RSA_private_decrypt(len, in, out, rsa, RSA_NO_PADDING); /* EXPECT:raw-rsa-openssl-no-padding-private */
    RSA_private_encrypt(len, in, out, rsa, RSA_NO_PADDING); /* EXPECT:raw-rsa-openssl-no-padding-private */
    EVP_PKEY_CTX_set_rsa_padding(ctx, RSA_NO_PADDING); /* EXPECT:raw-rsa-openssl-evp-no-padding */
    CK_MECHANISM mech = { CKM_RSA_X_509, NULL, 0 }; /* EXPECT:raw-rsa-pkcs11-ckm-rsa-x-509 */
}

void negative(RSA *rsa, EVP_PKEY_CTX *ctx, unsigned char *in, unsigned char *out, int len) {
    RSA_private_decrypt(len, in, out, rsa, RSA_PKCS1_OAEP_PADDING);
    RSA_public_encrypt(len, in, out, rsa, RSA_NO_PADDING);
    EVP_PKEY_CTX_set_rsa_padding(ctx, RSA_PKCS1_PSS_PADDING);
    CK_MECHANISM pss = { CKM_RSA_PKCS_PSS, NULL, 0 };
}
